### PR TITLE
external: format python files using tool black

### PR DESCRIPTION
**Pull request by subhamkrai**
`black` is python formatter tool which
automatically formats python files without
manual changing. just run `black <python-file>`
and done.

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
came across this nice tool `black` https://black.readthedocs.io/en/stable/ while working on #10374

**Comment by subhamkrai** [comment]
in follow up pr will be adding ci for this

**Comment by subhamkrai** [comment]
@travisn @parth-gr regarding our discussion, we can add `black` as the default formatting tool we just need to follow [this](https://dev.to/adamlombard/how-to-use-the-black-python-code-formatter-in-vscode-3lo0#:~:text=Black%20is%20%22the%20uncompromising%20Python,save%20a%20file%20in%20VSCode.&text=Open%20your%20VSCode%20settings%2C%20by,%3E%20Preferences%20%2D%3E%20Settings'.&text=Black%20will%20now%20format%20your,py%20file.) steps

**Comment by subhamkrai** [comment]
let me know how we want to proceed with this, do we want to add in the build step, or adding the above in the doc should be enough?

**Comment by parth-gr** [comment]
> `black` is python formatter tool which automatically formats python files without manual changing. just run `black <python-file>` and done.
>

Do we need to do it again and again whenever we would be adding commits to the python script locally?

**Comment by subhamkrai** [comment]
> > `black` is python formatter tool which automatically formats python files without manual changing. just run `black <python-file>` and done.
> >
> 
> Do we need to do it again and again whenever we would be adding commits to the python script locally?

not always  if we follow  the same format style when coding and also if we add the vscode setting just save the file and done. It will help.

Anyway  I'll  add check in ci which will fail if not formatted correctly.

File: deploy/examples/create-external-cluster-resources.py [hunk 1]
```diff
@@ -990,65 +1249,85 @@ def validate_rgw_endpoint(self):
...
+        self.out_map[
+            "RBD_METADATA_EC_POOL_NAME"
+        ] = self.validate_rgw_metadata_ec_pool_name()
+        self.out_map["RGW_POOL_PREFIX"] = self._arg_parser.rgw_pool_prefix
         if self._arg_parser.rgw_endpoint:
             if self._arg_parser.dry_run:
                 self.create_rgw_admin_ops_user()
             else:
-                self.out_map['RGW_ADMIN_OPS_USER_ACCESS_KEY'], self.out_map['RGW_ADMIN_OPS_USER_SECRET_KEY'], info_cap_supported = self.create_rgw_admin_ops_user()
+                (
```
Comment (parth-gr) [inline deploy/examples/create-external-cluster-resources.py:1315]: Hey @subhamkrai I am not sure about this formatting(of using () and separate lines just for variables) and best practices but this type of formatting makes me more difficult to understand the code.

**Comment by parth-gr** [comment]
I suppose `pylint` is the best for formatting and is used in most of the projects.

File: deploy/examples/create-external-cluster-resources.py (hunk 1 above)
Comment (subhamkrai) [reply deploy/examples/create-external-cluster-resources.py:1315]: can you point from the original file which lines you are pointing?
if you pointing to this line then I see what change tool has brought we can discuss offline also if you want.

**Comment by subhamkrai** [comment]
> I suppose `pylint` is the best for formatting and is used in most of the projects.

well, this a good point for discussion but here is what I have found, projects are moving to black, and also black has 6x GH starts then pylint. 
1. https://learn.adafruit.com/improve-your-code-with-pylint/black
2.  https://github.com/microsoft/presidio/issues/317#issuecomment-620715678
3. https://www.libhunt.com/compare-pylint-vs-black

 That being said, no tool is perfect

**Pull request merged**